examples/pom.xml

```diff
@@ -67,5 +78,20 @@
       <artifactId>alluxio-core-common</artifactId>
       <version>${project.version}</version>
     </dependency>
+    <dependency>
+      <groupId>org.alluxio</groupId>
+      <artifactId>alluxio-core-client-hdfs</artifactId>
+      <version>${project.version}</version>
+    </dependency>
+
+    <!-- The above Internal dependencies can be replaced with the following external
```

Comment:
not sure if this is appropriate to leave in here? is there a reason to use the published dependency instead of the locally compiled ones? note that the root pom.xml defines the examples module, so it gets compiled with everything else. if your intended example is to use an external dependency, you may also have to provide additional instructions on how to compile only the examples module without any references to internal dependencies, otherwise this can be more confusing

Reply:
A developer may choose to use Alluxio binary packages, and build their programs with the binary packages.
This is similar to those apache Hadoop binary packages.

>provide additional instructions on how to compile only the examples module without any references to internal dependencies, otherwise this can be more confusing

This is already documented in "docs/en/api/Java-API.md".
And there are some code snippet to show how to do that. These examples are compilable and ready to be run.

Reply:
i was imagining that the user would only need to compile the examples module instead of the entire repository. the compilation instruction is the documentation that i do not see in the Java-API doc page

Reply:
Yes. A developer would only need to compile these examples modules.

As of the current patch, these examples can be compiled within the entire repository, __AND__ they can also be compiled in a standalone way.

Reply:
i disagree that the current patch allows the examples module to be compiled in a standalone way because it is dependent on the parent alluxio modules. i would reverse the commented sections in this pom.xml file:
- uncomment this commented dependency on alluxio-shaded-client
- comment out the 2 internal alluxio dependencies

with these 2 actions + the independent hadoop version, the examples module is completely standalone. right now the current patch is standalone hadoop version but dependent on alluxio-core-common and alluxio-core-client-hdfs